### PR TITLE
Budget app

### PR DESCRIPTION
**Pull request by Uzair-Manzoor**
## This pull request has been created against below mentioned amendments into the project as per previous viewer's feedback:

- [x] Fixed issue with the create method to work as expected. Previously, It was crashing the app when user tried to add a category and was complaining about the icon being passed into it. The feature has been updated so the category can be created successfully and other features are testable.
- [x] Recorded a [presentation](https://www.loom.com/share/e06a206fb2d1466589eff6f915bede6d?sid=6ab2daed-7b7b-430b-b754-df37d89cc538) of the app and provided the link in the readme file.
- [x] Deployed the app on [render](https://budget-app2-4hbs.onrender.com/) for having the online version and provided the link in the readme file

### CODE REVIEW

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Uzair-Manzoor,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- App layout follows Behance design ✔️ 
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
- [ ] Kindly record a presentation of your app and provide the link in the readme file

- [ ] Kindly ensure your live demo is functional. When I try click the sign up button on the splash button to sign a user, the app throws error. Kindly fix the live demo app.

  image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -45,9 +45,9 @@ Ruby on Rails, Visual Studia Code, WSL
 
 - [Live Demo Link]()
```
Comment (binhussen) [inline README.md]: - Kindly, having the online version is not optional, please fix what the previous reviewer suggested and add it to fit the requirement and have better user accessibility :+1:

File: README.md [hunk 2]
```diff
@@ -45,9 +45,9 @@ Ruby on Rails, Visual Studia Code, WSL
...
 
-## 🚀 Presentation Video <a name="video-presentation"></a>
+## 🚀 Presentation <a name="video-presentation"></a>
 
-- [Presentation Link]()
+- [Presentation Link](https://www.loom.com/share/e06a206fb2d1466589eff6f915bede6d?sid=6ab2daed-7b7b-430b-b754-df37d89cc538)
```
Comment (binhussen) [inline README.md]: - Great job for adding the clear video presentation, I think you forgot to  highlight some interesting pieces of code as mentioned in the requirmnet :+1:

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @Uzair-Manzoor , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Highlights

- Sticked the given design ✔️
- Well stractured Code ✔️
- Used Git flow ✔️


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AmaduKamara — APPROVED** [review]
# Status: APPROVED 🟢 🟢 🟢

Hi @Uzair-Manzoor,

Good job working on the changes highlighted by the previous reviewer 👍🏽

Your project is complete! There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You set up Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Implement Signup and Login functionalities correctly 👍🏽
✅ Categories and Transactions are created successfully as expected 👍🏽
✅ Good descriptive README file 👍🏽
✅ Implemented tests correctly and all test cases passed 👍🏽
✅ The app works fine on the local server and the deployed one as well 👍🏽

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**